codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -58,45 +191,44 @@ public HttpContentCompressor(int compressionLevel) {
      * Creates a new handler with the specified compression level, window size,
      * and memory level..
      *
-     * @param compressionLevel
-     *        {@code 1} yields the fastest compression and {@code 9} yields the
-     *        best compression.  {@code 0} means no compression.  The default
-     *        compression level is {@code 6}.
-     * @param windowBits
-     *        The base two logarithm of the size of the history buffer.  The
-     *        value should be in the range {@code 9} to {@code 15} inclusive.
-     *        Larger values result in better compression at the expense of
-     *        memory usage.  The default value is {@code 15}.
-     * @param memLevel
-     *        How much memory should be allocated for the internal compression
-     *        state.  {@code 1} uses minimum memory and {@code 9} uses maximum
-     *        memory.  Larger values result in better and faster compression
-     *        at the expense of memory usage.  The default value is {@code 8}
+     * @param compressionLevel {@code 1} yields the fastest compression and {@code 9} yields the
+     *                         best compression.  {@code 0} means no compression.  The default
+     *                         compression level is {@code 6}.
+     * @param windowBits       The base two logarithm of the size of the history buffer.  The
+     *                         value should be in the range {@code 9} to {@code 15} inclusive.
+     *                         Larger values result in better compression at the expense of
+     *                         memory usage.  The default value is {@code 15}.
+     * @param memLevel         How much memory should be allocated for the internal compression
+     *                         state.  {@code 1} uses minimum memory and {@code 9} uses maximum
+     *                         memory.  Larger values result in better and faster compression
+     *                         at the expense of memory usage.  The default value is {@code 8}
+     * @deprecated Use the static factory method {@link #create()} and the <tt>with...</tt> methods to create
+     * and configure a new content compressor. Note: Creating a content compressor via <tt>create</tt> will
+     * apply a content length and content type filter, along with check which only compresses FullHttpResponses
+     * or ones with {@code "Transfer-Encoding: chunked"}.
      */
+    @Deprecated
```

Comment:
...because when using one of the existing constructors, the behaviour of the content compressor remains unchanged (backwards compatible). Only if you use the new factory method (yes I'd prefer a language construct as well, but as I wrote earlier, the default constructor needs to remain backward compatible as well) the content compressor will filter on content type, length and request type.

We could also not deprecate them, the intention is to push a new user of the class into the right direction, while providing existing user with backward compatibility and yet pointing out that there is a change to be considered. Most of the time it should be just fine to activate the newly implemented filters.